### PR TITLE
superhtml: init at 0.5.0

### PR DESCRIPTION
**Pull request by petertriho**
## Description of changes

Adds [superhtml](https://github.com/kristoff-it/superhtml) https://github.com/NixOS/nixpkgs/issues/340482

I don't use zig and have never packaged any zig projects before so I'm not sure if this the correct way to do things.

`zon2nix` doesn't work with `superhtml` so I just manually made the `deps.nix` file.

@matdibu @callumio 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by petertriho** [comment]
Switched to `fetchgit` for deps that were original git in `build.zig.zon` files

File: pkgs/by-name/su/superhtml/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  zig,
+}:
+stdenv.mkDerivation rec {
+  pname = "superhtml";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "kristoff-it";
+    repo = "superhtml";
+    rev = "v${version}";
```
Comment (NotAShelf) [inline pkgs/by-name/su/superhtml/package.nix:15]: `refs/tags/v${version}` would be preferred here.

Comment (petertriho) [reply pkgs/by-name/su/superhtml/package.nix]: Fixed

**Review by matdibu — APPROVED** [review]

**Comment by petertriho** [comment]
@NotAShelf @wegank Is this anything else I need to do to get this merged?

File: pkgs/by-name/su/superhtml/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  postPatch = ''
+    ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
+  '';
+
+  meta = {
+    description = "HTML Language Server and Templating Language Library";
+    homepage = "https://github.com/kristoff-it/superhtml";
+    license = lib.licenses.mit;
+    mainProgram = "superhtml";
+    maintainers = with lib.maintainers; [ petertriho ];
```
Comment (wegank) [inline pkgs/by-name/su/superhtml/package.nix:32]:
```suggestion
    maintainers = with lib.maintainers; [ petertriho ];
    platforms = lib.platforms.unix;
```

Comment (petertriho) [reply pkgs/by-name/su/superhtml/package.nix]: @wegank Oops, thank you. Fixed

**Review by wegank — APPROVED** [review]

**Pull request merged**